src/api.ts

```diff
@@ -62,92 +63,105 @@ export class AuthenticationError extends HTTPError {
 }
 
 export class JamsocketApi {
-    apiBase: string
-
-    constructor(private auth?: string) {
-      this.apiBase = process.env.JAMSOCKET_SERVER_API ?? 'https://jamsocket.dev'
+  constructor(private apiBase: string, private options: https.RequestOptions = {}) {}
+
+  public static fromEnvironment(): JamsocketApi {
+    const override = process.env.JAMSOCKET_SERVER_API
+    let apiBase
+    if (override === undefined) {
+      apiBase = 'https://jamsocket.dev'
+    } else {
+      console.warn(`Using Jamsocket server override: ${override}`)
+      apiBase = override
     }
 
-    private async makeRequest(endpoint: string, method: HttpMethod, body?: any, headers?: Headers): Promise<any> {
-      const url = `${this.apiBase}${endpoint}`
-      const response = await request(url, body || null, { method, headers })
-
-      let responseBody
-      try {
-        responseBody = JSON.parse(response.body)
-      } catch (error) {
-        throw new Error(`jamsocket: error parsing JSON response - ${error}: ${response.body}`)
-      }
+    return new JamsocketApi(apiBase, {})
+  }
 
-      if (response.statusCode && response.statusCode >= 400) {
-        const { message, status, code, id } = responseBody.error
-        throw new HTTPError(response.statusCode, `jamsocket: ${status} - ${code}: ${message} (id: ${id})`)
-      }
+  private async makeRequest(endpoint: string, method: HttpMethod, body?: any, headers?: Headers): Promise<any> {
+    const url = `${this.apiBase}${endpoint}`
+    const response = await request(url, body || null, { method, headers, ...this.options })
```

Comment:
I think the `...this.options` should come before the `method` and `headers` so the values passed into `makeRequest` override the options provided at instantiation.